home-work-app/src/app/services/MovieService.ts

```diff
@@ -1,14 +1,76 @@
+import { forIn } from 'lodash';
+import { AppConstants } from '../configs/appConstants';
 import { IApiResponse } from '../interfaces/IApiResponse';
-import { movieResponse } from '../../../tempData/GetMoviesResponse';
+import { store } from '../redux/store';
 
 export class MovieService {
-    static async getMovies(): Promise<IApiResponse.GetMoviesResponse> {
-        const response: IApiResponse.GetMoviesResponse = await new Promise(
-            (resolve: (collection: IApiResponse.GetMoviesResponse) => void): void =>
-        {
-            setTimeout(() => {
-                resolve(movieResponse);
-            }, 4000);
+    static apiUrl: string = AppConstants.apiConfig.url;
+
+    static getRequestConfig(command?: string, data?: {[key: string]: string | number}): IApiResponse.RequestConfig {
+        let method: string;
+        switch (command) {
+            case 'createMovieItem': method = 'POST';
+            case 'removeItem': method = 'DELETE';
+            default: method = 'GET';
+        }
+
+        return {
+            method: 'GET',
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify(data)
+        };
+    }
+
+    static getURL(params?: {[key: string]: string | number | string[]}): string {
+        let url: string = this.apiUrl;
+        if (params && Object.keys(params).length) {
+            url = url + '?';
+            forIn(params, (value: string | number | string[] , key: string): void => {
+                let sortDirection: string = 'desc';
+                if (key === 'sortBy' && value === 'title') {
+                    sortDirection = 'acs';
+                }
+                url = url + `${key}=${value}&sortOrder=${sortDirection}&limit=20&`;
+            });
+        }
+
+        return url;
+    }
+
+    static async getMovies(params?: {[key: string]: string | number | string[]}):
+    Promise<IApiResponse.GetMoviesResponse> {
+        const response: IApiResponse.GetMoviesResponse = await fetch(
+            this.getURL(params),
+            this.getRequestConfig()
+        )
+        .then((fetchedData: Response): Promise<IApiResponse.GetMoviesResponse> => fetchedData.json());
```

Comment:
Since this is `async` function let's rework it in async/await style without `Promise.then`
 ```
const response: Response = await fetch(
     this.getURL(params),
     this.getRequestConfig()
);
const data: IApiResponse.GetMoviesResponse = await response.json();
```